test-data/unit/check-literal.test

```diff
@@ -2227,7 +2227,7 @@ d[c_key]                      # E: TypedDict "Outer" has no key 'c'
 
 reveal_type(d.get(a_key, u))  # N: Revealed type is 'Union[builtins.int, __main__.Unrelated]'
 reveal_type(d.get(b_key, u))  # N: Revealed type is 'Union[builtins.str, __main__.Unrelated]'
-d.get(c_key, u)               # E: TypedDict "Outer" has no key 'c'
+reveal_type(d.get(c_key, u))  # N: Revealed type is 'builtins.object'
```

Comment:
Shouldn't this infer `Unrelated`, or at least a Union of something with `Unrelated`?

Reply:
Like in the other discussion, the type here would be `Union[builtins.object, Unrelated]`, which mypy simplifies to `builtins.object`